connection.go

```diff
@@ -131,70 +152,25 @@ func (ct *ConnectionTransportTLS) Close() {
 	}
 }
 
-// SharedKeybaseTransport is a ConnectionTransport implementation that
-// uses a shared local socket to a keybase daemon.
-type SharedKeybaseTransport struct {
-	kbCtx *libkb.GlobalContext
-
-	// Protects everything below.
-	mutex           sync.Mutex
-	transport       rpc.Transporter
-	stagedTransport rpc.Transporter
-}
-
-// Test that SharedKeybaseTransport fully implements the
-// ConnectionTransport interface.
-var _ ConnectionTransport = (*SharedKeybaseTransport)(nil)
-
-// Dial is an implementation of the ConnectionTransport interface.
-func (kt *SharedKeybaseTransport) Dial(ctx context.Context) (
-	rpc.Transporter, error) {
-	_, transport, _, err := kt.kbCtx.GetSocket(true)
-	if err != nil {
-		return nil, err
-	}
-
-	kt.mutex.Lock()
-	defer kt.mutex.Unlock()
-	kt.stagedTransport = transport
-	return transport, nil
-}
-
-// IsConnected is an implementation of the ConnectionTransport interface.
-func (kt *SharedKeybaseTransport) IsConnected() bool {
-	kt.mutex.Lock()
-	defer kt.mutex.Unlock()
-	return kt.transport != nil && kt.transport.IsConnected()
-}
-
-// Finalize is an implementation of the ConnectionTransport interface.
-func (kt *SharedKeybaseTransport) Finalize() {
-	kt.mutex.Lock()
-	defer kt.mutex.Unlock()
-	kt.transport = kt.stagedTransport
-	kt.stagedTransport = nil
-}
-
-// Close is an implementation of the ConnectionTransport interface.
-func (kt *SharedKeybaseTransport) Close() {
-	// Since this is a shared connection, do nothing.
-}
+type LogTagsFromContext func(ctx context.Context) (map[interface{}]string, bool)
```

Comment:
1) We don't lose it, it will still be in the original repo.  In my opinion it's not generic enough to be in the rpc repo, as it's specific to trying to share a transport among multiple Keybase daemon connections.

2) These tags aren't the RPC tags, they will be used in practice to transfer logger tags into RPC-land: https://github.com/keybase/client/blob/master/go/logger/standard.go#L40.  The key needs to be an interface, to avoid having to coordinate unique keys among all the applications who might concurrently be sharing a context with debug tags.